CI failure in unit tests:

```
E     File "/root/src/github.com/ansible/ansible/lib/ansible/modules/network/nxos/nxos_bgp_neighbor_af.py", line 591
E       commands.append('{0} {2} {1}'.format(*key.split(), value))
E   SyntaxError: only named arguments may follow *expression
```

--- a/lib/ansible/modules/network/nxos/nxos_bgp_neighbor_af.py
+++ b/lib/ansible/modules/network/nxos/nxos_bgp_neighbor_af.py
@@ -628,16 +628,12 @@ def state_present(module, existing, proposed, candidate):
 
         parents.append('neighbor {0}'.format(module.params['neighbor']))
 
-        if len(commands) == 1:
-            candidate.add(commands, parents=parents)
-        elif len(commands) > 1:
-            af_command = 'address-family {0} {1}'.format(
-                module.params['afi'], module.params['safi'])
-            if af_command in commands:
-                commands.remove(af_command)
-                parents.append('address-family {0} {1}'.format(
-                    module.params['afi'], module.params['safi']))
-                candidate.add(commands, parents=parents)
+        af_command = 'address-family {0} {1}'.format(
+            module.params['afi'], module.params['safi'])
+        parents.append(af_command)
+        if af_command in commands:
+            commands.remove(af_command)
+        candidate.add(commands, parents=parents)
 
 
 def state_absent(module, existing, candidate):

--- a/lib/ansible/modules/network/nxos/nxos_bgp_neighbor_af.py
+++ b/lib/ansible/modules/network/nxos/nxos_bgp_neighbor_af.py
@@ -694,9 +694,6 @@ def main():
     module = AnsibleModule(
         argument_spec=argument_spec,
         mutually_exclusive=[['advertise_map_exist', 'advertise_map_non_exist']],
-        required_together=[['max_prefix_limit', 'max_prefix_interval'],
-                           ['max_prefix_limit', 'max_prefix_warning'],
-                           ['max_prefix_limit', 'max_prefix_threshold']],
         supports_check_mode=True,
     )
 
@@ -705,7 +702,11 @@ def main():
     result = dict(changed=False, warnings=warnings)
 
     state = module.params['state']
-
+    for key in ['max_prefix_interval', 'max_prefix_warning', 'max_prefix_threshold']:
+        if module.params[key] and not module.params['max_prefix_limit']:
+            module.fail_json(
+                msg='max_prefix_limit is required when using %s' % key
+            )
     if module.params['vrf'] == 'default' and module.params['soo']:
         module.fail_json(msg='SOO is only allowed in non-default VRF')
 

--- a/test/units/modules/network/nxos/fixtures/nxos_bgp/config.cfg
+++ b/test/units/modules/network/nxos/fixtures/nxos_bgp/config.cfg
@@ -9,3 +9,4 @@ router bgp 65535
     timers bgp 1 10
   neighbor 3.3.3.5
     address-family ipv4 unicast
+      maximum-prefix 30 30

--- a/test/units/modules/network/nxos/test_nxos_bgp_neighbor_af.py
+++ b/test/units/modules/network/nxos/test_nxos_bgp_neighbor_af.py
@@ -70,7 +70,7 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
                              advertise_map_exist=['my_advertise_map', 'my_exist_map']))
         self.execute_module(
             changed=True, sort=False,
-            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'advertise-map my_advertise_map exist my_exist_map']
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'address-family ipv4 unicast', 'advertise-map my_advertise_map exist my_exist_map']
         )
 
     def test_nxos_bgp_neighbor_af_advertise_map_non_exist(self):
@@ -78,7 +78,7 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
                              advertise_map_non_exist=['my_advertise_map', 'my_exist_map']))
         self.execute_module(
             changed=True, sort=False,
-            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'advertise-map my_advertise_map non-exist my_exist_map']
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'address-family ipv4 unicast', 'advertise-map my_advertise_map non-exist my_exist_map']
         )
 
     def test_nxos_bgp_neighbor_af_max_prefix_limit_default(self):
@@ -86,7 +86,7 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
                              safi='unicast', max_prefix_limit='default'))
         self.execute_module(
             changed=True, sort=False,
-            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'no maximum-prefix']
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'address-family ipv4 unicast', 'no maximum-prefix']
         )
 
     def test_nxos_bgp_neighbor_af_max_prefix(self):
@@ -95,5 +95,5 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
                              max_prefix_limit=20))
         self.execute_module(
             changed=True, sort=False,
-            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'maximum-prefix 20 20']
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'address-family ipv4 unicast', 'maximum-prefix 20 20']
         )

--- a/test/units/modules/network/nxos/test_nxos_bgp_neighbor_af.py
+++ b/test/units/modules/network/nxos/test_nxos_bgp_neighbor_af.py
@@ -42,7 +42,7 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
         self.mock_get_config.stop()
 
     def load_fixtures(self, commands=None, device=''):
-        self.get_config.return_value = load_fixture('', 'nxos_bgp_config.cfg')
+        self.get_config.return_value = load_fixture('nxos_bgp', 'config.cfg')
         self.load_config.return_value = []
 
     def test_nxos_bgp_neighbor_af(self):
@@ -86,5 +86,14 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
                              safi='unicast', max_prefix_limit='default'))
         self.execute_module(
             changed=True, sort=False,
-            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'maximum-prefix default']
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'no maximum-prefix']
+        )
+
+    def test_nxos_bgp_neighbor_af_max_prefix(self):
+        set_module_args(dict(asn=65535, neighbor='3.3.3.5', afi='ipv4',
+                             safi='unicast', max_prefix_threshold=20,
+                             max_prefix_limit=20))
+        self.execute_module(
+            changed=True, sort=False,
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'maximum-prefix 20 20']
         )

--- a/test/units/modules/network/nxos/test_nxos_bgp_neighbor_af.py
+++ b/test/units/modules/network/nxos/test_nxos_bgp_neighbor_af.py
@@ -80,3 +80,11 @@ class TestNxosBgpNeighborAfModule(TestNxosModule):
             changed=True, sort=False,
             commands=['router bgp 65535', 'neighbor 3.3.3.5', 'advertise-map my_advertise_map non-exist my_exist_map']
         )
+
+    def test_nxos_bgp_neighbor_af_max_prefix_limit_default(self):
+        set_module_args(dict(asn=65535, neighbor='3.3.3.5', afi='ipv4',
+                             safi='unicast', max_prefix_limit='default'))
+        self.execute_module(
+            changed=True, sort=False,
+            commands=['router bgp 65535', 'neighbor 3.3.3.5', 'maximum-prefix default']
+        )

